cli/internal/runcache/runcache.go

```diff
@@ -106,9 +106,27 @@ type TaskCache struct {
 	LogFileName       turbopath.AbsoluteSystemPath
 }
 
+// IsCacheHit returns true if we have a cache hit, false if we have a cache miss
+func (tc TaskCache) IsCacheHit(ctx context.Context) bool {
+	if tc.cachingDisabled || tc.rc.readsDisabled {
+		return false
+	}
+	changedOutputGlobs, err := tc.rc.outputWatcher.GetChangedOutputs(ctx, tc.hash, tc.repoRelativeGlobs.Inclusions)
+	if err != nil {
+		changedOutputGlobs = tc.repoRelativeGlobs.Inclusions
+	}
+
+	hasChangedOutputs := len(changedOutputGlobs) > 0
+	if hasChangedOutputs {
+		hitStatus := tc.rc.cache.Exists(tc.hash)
+		return hitStatus.Local || hitStatus.Remote
+	}
+
+	return true
+}
+
 // RestoreOutputs attempts to restore output for the corresponding task from the cache.
-// Returns true if successful.
-func (tc TaskCache) RestoreOutputs(ctx context.Context, prefixedUI *cli.PrefixedUi, progressLogger hclog.Logger) (bool, error) {
+func (tc TaskCache) RestoreOutputs(ctx context.Context, prefixedUI *cli.PrefixedUi, progressLogger hclog.Logger, logStatusOnly bool) (bool, error) {
```

Comment:
I don't think we ought to have a mode of `RestoreOutputs` where the outputs are not restored. I suspect we can handle log output ordering without affecting the cache and file IO of `turbo`

Reply:
I had to be able to display the logs for cache misses seperate from the actual restoring. These are highly intertwined and therefore this was necessary. It won't be necessary in the queue based implementation.